libs/metagen/src/client_py/utils.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use heck::*;
+use once_cell::sync::Lazy;
+
+// sourced from
+// keyword library https://docs.python.org/3/library/keyword.html
+static KEYWORDS: Lazy<std::collections::HashSet<&'static str>> = Lazy::new(|| {
+    [
+        "False", "None", "True", "and", "as", "assert", "async", "await", "break", "class",
+        "continue", "def", "del", "elif", "else", "except", "finally", "for", "from", "global",
+        "if", "import", "in", "is", "lambda", "nonlocal", "not", "or", "pass", "raise", "return",
+        "try", "while", "with", "yield",
+    ]
+    .into_iter()
```

Comment:
Nice catch! reserved keyword checks are definitely lacking on the metagen py side.
Or instead allow it by normalizing into something else..
https://discuss.python.org/t/should-python-allow-reserved-keywords-as-property-names/39212/17

Reply:
The new implementation doesn't also handle them...yet. It's way more trickier compared to the rust/ts approaches. Ultimately, we'll need to find a way to add decorators/metadata to the fields/classes and interject during serialization. At least that's the only viable way I've found so far.

Reply:
We can also rely on the alternate TypeDict [syntax](https://docs.python.org/3/library/typing.html#typing.TypedDict). It looks something like:

```py
# Alternative syntax
Point2D = TypedDict('Point2D', {'x': int, 'y': int}, total=False)
```

I'm unsure about because your `Struct` implementation seems to be dong a lot more. Do you think we can replace it on a system based on TypeDicts?

Reply:
I see, the main goal was to have static type hints so I don't think it will work well.. 🤔